### PR TITLE
feat(server): memory includes partners assets on timeline

### LINKED ISSUE #7991: Memories Bar not shown for secondary users
### The bug

When I log in as my account, I do see a memories bar at the top of both the web and IOS display. When I look at any other user on the server they do not have the same bar. This applies to shared libraries and non-shared libraries.

When I log in, I see a bar at the top of both the web interface and the IOS app. When I login as my wife, who I share the library with she does not have that as part of her display. When I login as any other user that it have their own library (ie I am not sharing with them and they are not sharing with anyone else) they also do not have a display of their memories.

### The OS that Immich Server is running on

Ubuntu 22.04.4 LTS

### Version of Immich Server

1.98.2

### Version of Immich Mobile App

1.98.2 build.144

### Platform with the issue

- [ ] Server
- [X] Web
- [X] Mobile

### Your docker-compose.yml content

```YAML
version: "3.8"

#
# WARNING: Make sure to use the docker-compose.yml of the current release:
#
# https://github.com/immich-app/immich/releases/latest/download/docker-compose.yml
#
# The compose file on main may not be compatible with the latest release.
#

name: immich

services:
  immich-server:
    container_name: immich_server
    image: ghcr.io/immich-app/immich-server:${IMMICH_VERSION:-release}
    command: [ "start.sh", "immich" ]
    volumes:
      - ${UPLOAD_LOCATION}:/usr/src/app/upload
      - /etc/localtime:/etc/localtime:ro
    env_file:
      - .env
    ports:
      - 2283:3001
    depends_on:
      - redis
      - database
    restart: always

  immich-microservices:
    container_name: immich_microservices
    image: ghcr.io/immich-app/immich-server:${IMMICH_VERSION:-release}
    # extends:
    #   file: hwaccel.yml
    #   service: hwaccel
    command: [ "start.sh", "microservices" ]
    volumes:
      - ${UPLOAD_LOCATION}:/usr/src/app/upload
      - /etc/localtime:/etc/localtime:ro
    env_file:
      - .env
    depends_on:
      - redis
      - database
    restart: always

  immich-machine-learning:
    container_name: immich_machine_learning
    image: ghcr.io/immich-app/immich-machine-learning:${IMMICH_VERSION:-release}
    volumes:
      - model-cache:/cache
    env_file:
      - .env
    restart: always

  redis:
    container_name: immich_redis
    image: redis:6.2-alpine@sha256:afb290a0a0d0b2bd7537b62ebff1eb84d045c757c1c31ca2ca48c79536c0de82
    restart: always

  database:
    container_name: immich_postgres
    image: tensorchord/pgvecto-rs:pg14-v0.2.0@sha256:90724186f0a3517cf6914295b5ab410db9ce23190a2d9d0b9dd6463e3fa298f0
    env_file:
      - .env
    environment:
      POSTGRES_PASSWORD: ${DB_PASSWORD}
      POSTGRES_USER: ${DB_USERNAME}
      POSTGRES_DB: ${DB_DATABASE_NAME}
    volumes:
      - pgdata:/var/lib/postgresql/data
    restart: always

volumes:
  pgdata:
  model-cache:
```

### Your .env content

```Shell
# You can find documentation for all the supported env variables at https://immich.app/docs/install/environment-variables

# The location where your uploaded files are stored
UPLOAD_LOCATION=/immich

# The Immich version to use. You can pin this to a specific version like "v1.71.0"
IMMICH_VERSION=release

# Connection secrets for postgres and typesense. You should change these to random passwords
TYPESENSE_API_KEY=[redact]
DB_PASSWORD=[redact]

# The values below this line do not need to be changed
###################################################################################
DB_HOSTNAME=immich_postgres
DB_USERNAME=[redact]
DB_DATABASE_NAME=immich

REDIS_HOSTNAME=immich_redis
```

### Reproduction steps

```bash
1.With running server consisting of photos
2.Login to web as admin
3. note 'memories' display at top of page
4. log out
5. login as another user
6. note missing memories from this user's page

on IOS
7. login as second user
8. note missing memories section at top of app.
...
```

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by alextran1502**
fixes #7991

Show partners assets on memory lane if show in timeline is enabled

We have a similar method in `searchService` i.e `getUserIdsToSearch` but I am not sure where to put it to be able to reuse code

### CODE REVIEW

File: server/src/domain/asset/asset.service.ts [hunk 1]
```diff
@@ -5,6 +5,7 @@ import _ from 'lodash';
 import { DateTime, Duration } from 'luxon';
 import { extname } from 'node:path';
 import sanitize from 'sanitize-filename';
+import { SearchService } from '..';
```
Comment (danieldietzler) [inline server/src/domain/asset/asset.service.ts:8]: This is a bad import. Could you import the folder or file directly?

**Review by danieldietzler — COMMENTED** [review]
Maybe you could move the duplicate code to a method in `domain.util`? Alternatively, since it is partner-related, it could be a method there

**Review by jrasm91 — APPROVED** [review]
Maybe we should move this to a dedicated sub select in the asset builder instead.

**Comment by alextran1502** [comment]
@jrasm91 We use a pretty specific query to get the date group for memories. I think creating a sub-select query for asset builder might not help in this case.

image

**Pull request merged**